inc/t_cose/t_cose_sign1_sign.h

```diff
@@ -148,6 +162,40 @@ t_cose_sign1_sign_init(struct t_cose_sign1_sign_ctx *context,
                        int32_t                       cose_algorithm_id);
 
 
+/**
+ * \brief  Set the restart context for restartable signing.
+ *
+ * \param[in] context            The t_cose signing context.
+ * \param[in] rst_ctx            The t_cose restartable context.
+ *
+ * Initialize the \ref t_cose_sign1_sign_restart_ctx rst_ctx.
+ *
+ * Setting the \ref t_cose_sign1_sign_restart_ctx rst_ctx with this function
+ * will cause the function \ref t_cose_sign1_sign to execute the signing in a
+ * restartable manner.
+ */
+enum t_cose_err_t
+t_cose_sign1_set_restart_context(
+                        struct t_cose_sign1_sign_ctx         *context,
+                        struct t_cose_sign1_sign_restart_ctx *rst_ctx);
```

Comment:
I think set_restart_context is a good idea, but I think rst_ctx should be a void *. That should results in no need for t_cose_crypto_public.h and simplify a whole mess of dependencies. 

The t_cose implementation itself doesn't need to know how big this structure is. It never allocates it on the stack or copies it. It only passes a pointer to it around.

The lack of strong typing here won't be a huge problem. People won't mix it up much and even if it is a little bit of an issue the reduction of dependency in the public interface used by all is worth it.

I working on the assumption here that restart is seldom used.

The caller that does use it will have to allocate the context of the right type and size for the restartable crypto and initialize it. It will then pass through t_cose as a void *.